SingularityUI/app/components/taskDetail/TaskDetail.jsx

```diff
@@ -180,6 +180,10 @@ class TaskDetail extends Component {
 
         file.fullPath = `${files.fullPathToRoot}/${files.currentDirectory}/${file.name}`;
         file.downloadLink = `${config.apiRoot}/tasks/download?slaveHostname=${files.slaveHostname}&path=${file.fullPath}`;
+        const extensionMatcher = /(?:\.([^.]+))?$/;
```

Comment:
What about using something like `FilenameUtils.getExtension(fileName);` instead of a regex for clarity?

Reply:
java(script) :lol:

Reply:
Never mind, javascript :facepalm: